plugins/vcenter_logs.yaml

```diff
@@ -0,0 +1,70 @@
+version: 0.0.1
+title: VMware vCenter
+description: Log parser for VMware vCenter
+parameters:
+  - name: listen_port
+    type: int
+    default: 5142
+  - name: listen_ip
+    type: string
+    default: "0.0.0.0"
+  - name: max_buffer_size
+    type: string
+    default: "1024kib"
+    required: false
+  - name: enable_tls
+    type: bool
+    default: false
+  - name: certificate_file
+    type: string
+    default: "/opt/cert"
+  - name: key_file
+    type: string
+    default: "/opt/key"
+
+template: |
+  receivers:
+    tcplog:
+      listen_address: '{{ .listen_ip }}:{{ .listen_port }}'
+      max_log_size: {{ .max_buffer_size }}
+      attributes:
+        log_type: vmware_vcenter
+      add_attributes: true
+      {{ if .enable_tls }}
+      tls:
+        cert_file: {{ .certificate_file }}
+        key_file: {{ .key_file }}
+      {{ end }}
+      operators:
+        # vcenter will (sometimes) prepend an id to the messages, check
+        # for the id and drop it if it exsits
+        # example: '257 <14>1. . . '
+        - id: prefix_router
+          type: router
+          routes:
+            - output: pre_parser
+              expr: 'body matches "^\\d* "'
+          default: vcenter_parser
+
+        - id: pre_parser
+          type: regex_parser
+          regex: '^(?P<drop>\d* )(?P<syslog_message>[\w\W]*)'
+
+        - id: remove_drop
+          type: remove
+          field: attributes.drop
+        
+        - id: move_syslog_message
+          type: move
+          from: attributes.syslog_message
+          to: attributes
```

Comment:
this won't work I think you probably want this to go to `body`

Reply:
so do operators resolve in order? so in this case we have a router which can 'jump' the sequential order for one case to the very last operator. but otherwise we go through each operator? this is just a knowledge question for me

Reply:
@StefanKurek 

> so do operators resolve in order? so in this case we have a router which can 'jump' the sequential order for one case to the very last operator. but otherwise we go through each operator? this is just a knowledge question for me

Yes, they do. If the `output` field on an operator isn't specified it just goes to the one below it. In the case of a router you'll need something to tie the split back together.

Reply:
@armstrmi That being said can you set the `output` of this to the `vcenter_parser` just to be explicit?

Reply:
@cpheps why set the output of just this one and not any of the ones earlier in the chain? i almost feel like that makes it more arbitrary/confusing. just my opinion.